README.md

```diff
@@ -1,47 +1,178 @@
 # Pulumi Native Provider Boilerplate
 
-This repo is a boilerplate showing how to create a native Pulumi provider.  
+This repository is a boilerplate showing how to create a native Pulumi provider.
 
 ### Background
 
 This repository is part of the [guide for authoring and publishing a Pulumi Package](https://www.pulumi.com/docs/guides/pulumi-packages/how-to-author).
 
 Learn about the concepts behind [Pulumi Packages](https://www.pulumi.com/docs/guides/pulumi-packages/#pulumi-packages).
 
-## Creating a Pulumi Native Provider
+See where Resource Providers fit into [the Pulumi architecture](https://www.pulumi.com/docs/intro/concepts/how-pulumi-works/#how-pulumi-works)
 
-You can search-replace `xyz` with the name of your desired provider as a starting point for creating a provider that manages resources in the target cloud.
+A Pulumi Resource Provider: 
+- is a gRPC server which allows for the Pulumi engine to create resources in a specific cloud
+- holds the lifecycle logic for these cloud resources
+- holds a pulumi JSON schema that describes the provider
+- provides language-specific SDKs so resources can be created in whichever language you prefer
 
-Most of the code for the provider implementation is in `pkg/provider/provider.go`.  
+When we speak of a "native" provider, we mean that all implementation is native to Pulumi, as opposed 
+to [Terraform based providers](https://github.com/pulumi/pulumi-tf-provider-boilerplate).
 
-An example of using the single resource defined in this example is in `examples/simple`.
+## Authoring a Pulumi Native Provider
 
-A code generator is available which generates SDKs in TypeScript, Python, Go and .NET which are also checked in to the `sdk` folder.  The SDKs are generated from a schema in `provider/cmd/pulumi-resource-xyz/schema.json`.  This file should be kept aligned with the resources, functions and types supported by the provider implementation.
+The following instructions cover providers maintained by Pulumi (denoted with a "Pulumi Official" checkmark on the Pulumi registry).
```

Comment:
I think we can just say "this guide assumes that the provider is written for the pulumi organisation